timeout_handler.go

```diff
@@ -0,0 +1,231 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package simplex
+
+import (
+	"container/heap"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+type TimeoutTask struct {
+	NodeID   NodeID
+	TaskID   string
+	Start    uint64
+	End      uint64
+	Task     func()
+	Deadline time.Time
+
+	index int // for heap to work more efficiently
+}
+
+type TimeoutHandler struct {
+	lock sync.Mutex
+
+	ticks chan time.Time
+	close chan struct{}
+	// nodeids -> range -> task
+	tasks map[string]map[string]*TimeoutTask
```

Comment:
isn't it simpler to just have:

`tasks map[string]*TimeoutTask 

where the key is:

```
func getTimeoutID(node NodeID, start, end uint64) string {
	return fmt.Sprintf("%x%d%s%d", node, start, delimiter, end)
}
```

? 

saves you an indirection and simplifies the code, in my opinion.